rclcpp_lifecycle/src/state.cpp

```diff
@@ -75,4 +102,48 @@ State::label() const
   return state_handle_->label;
 }
 
+void
+free_resources(State & instance)
+{
+  // nothing to free here
+  if (!instance.state_handle_) {
+    return;
+  }
+  if (!instance.owns_rcl_state_handle_) {
+    return;
+  }
+
+  if (instance.state_handle_) {
+    if (instance.state_handle_->label) {
+      instance.allocator_.deallocate(
+        const_cast<char *>(instance.state_handle_->label),
+        instance.allocator_.state);
+    }
+    instance.allocator_.deallocate(
+      const_cast<rcl_lifecycle_state_t *>(instance.state_handle_),
+      instance.allocator_.state);
```

Comment:
This should set `state_handle_` to `nullptr`. Even in the cases of the early return.

Maybe `owns_rcl_state_handle_` should be reset too.